tests/js/sentry-test/performance/utils.ts

```diff
@@ -86,33 +85,41 @@ export class TransactionEventBuilder {
     };
   }
 
-  addSpan(mSpan: MockSpan, parentSpanId?: string) {
+  generateSpanId() {
     // Convert the num of spans to a hex string to get its ID
-    const spanId = (this.#spans.length + 1).toString(16).padStart(16, '0');
-    const {span} = mSpan;
-    span.span_id = spanId;
-    span.trace_id = this.TRACE_ID;
-    span.parent_span_id = parentSpanId ?? this.ROOT_SPAN_ID;
-
-    this.#event.entries[0].data.push(span);
-
-    switch (mSpan.problemSpan) {
-      case ProblemSpan.PARENT:
-        this.#event.perfProblem?.parentSpanIds.push(spanId);
-        break;
-      case ProblemSpan.OFFENDER:
-        this.#event.perfProblem?.offenderSpanIds.push(spanId);
-        break;
-      default:
-        break;
-    }
+    return (this.#spans.length + 1).toString(16).padStart(16, '0');
+  }
 
-    if (span.timestamp > this.#event.endTimestamp) {
-      this.#event.endTimestamp = span.timestamp;
+  addSpan(mSpan: MockSpan, numSpans = 1, parentSpanId?: string) {
+    for (let i = 0; i < numSpans; i++) {
+      const spanId = this.generateSpanId();
+      const {span} = mSpan;
+      const clonedSpan = {...span};
+
+      clonedSpan.span_id = spanId;
+      clonedSpan.trace_id = this.TRACE_ID;
+      clonedSpan.parent_span_id = parentSpanId ?? this.ROOT_SPAN_ID;
+
+      this.#spans.push(clonedSpan);
+
+      switch (mSpan.problemSpan) {
+        case ProblemSpan.PARENT:
+          this.#event.perfProblem?.parentSpanIds.push(spanId);
```

Comment:
Might have missed this on another PR but i'm curious on this `#` format, looks a bit funky to me

Reply:
Ohh, very interesting! I did not know that, was curious why you didn't use `private` keyword in typescript as imo it looks a lot cleaner. But I'm assuming its for the reasons outline here
https://stackoverflow.com/questions/59641564/what-are-the-differences-between-the-private-keyword-and-private-fields-in-types

Reply:
Yep exactly, although it doesn't really matter here since they accomplish the same thing since it's just used for tests